### DISCUSSION TITLE
Using the SHT30 Temp/Humid sensor

### DISCUSSION THREAD

**Discussion by seamusdemora** [WiringPi/WiringPi#249] [Q&A]
**Objective:** I want to control an [SHT30 temperature/humidity sensor](https://www.adafruit.com/product/5064) using my Raspberry Pi. I'd much prefer to use WiringPi for this (instead of Python - or something else). 

I have looked through the [Examples section](https://github.com/WiringPi/WiringPi/tree/master/examples), and found the [`rht03.c` example](https://github.com/WiringPi/WiringPi/blob/master/examples/rht03.c). It seems the function call used to read the register over the I2C bus is `readRHT03`. I tried to find where this function was defined: I looked in [`wiringPi.h`](https://github.com/WiringPi/WiringPi/blob/master/wiringPi/wiringPi.h), but did not find the definition of `readRHT03` ;  I was unable to find the `maxdetect.h` file. 

Can someone clue me in on what I'm missing?  Is my objective a reasonable one? 

N.B.: I don't have much experience in C, so I may be way off base here; most of my C experience is with the Pico. I've written a couple of C programs that use the I2C bus for communicating with "similar" devices (e.g. [INA260](https://www.ti.com/product/INA260)).

**Comment by mstroh76** [comment]
I think the RHT03 (DHT22) is a different sensor and has nothing to do with SHT30.
I found in datasheet "I2C Interface with communication speeds up to 1 MHz and two user selectable addresses". So what you need is a I2C example.

**Reply by seamusdemora** [reply]
OK - can you point me toward an I2C example?
